File: php/src/Model/AcceptWorkerInviteResult.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * What a worker gets back after consuming an emailed invite. Unlike {@see JoinWorkspaceResult} the worker id already existed — an operator created the identity when they sent the invite.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class AcceptWorkerInviteResult
{
    public function __construct(
        public readonly string $token,
        public readonly string $workspaceId,
        public readonly string $workerId,
        public readonly string $label,
        public readonly ?string $expiresAt,
        public readonly string $portalUrl,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            token: (string) ($data['token'] ?? ''),
            workspaceId: (string) ($data['workspaceId'] ?? ''),
            workerId: (string) ($data['workerId'] ?? ''),
            label: (string) ($data['label'] ?? ''),
            expiresAt: isset($data['expiresAt']) ? (string) $data['expiresAt'] : null,
            portalUrl: (string) ($data['portalUrl'] ?? ''),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/AcceptWorkerInviteResult.java
package io.fivexer.sdk.model;

/**
 * What a worker gets back after consuming an emailed invite. Unlike {@link JoinWorkspaceResult} the worker id already existed — an operator created the identity when they sent the invite.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class AcceptWorkerInviteResult {
    private String token;
    private String expiresAt;
    private String workspaceId;
    private String workerId;
    private String label;
    private String portalUrl;

    public String getToken() { return token; }
    public String getExpiresAt() { return expiresAt; }
    public String getWorkspaceId() { return workspaceId; }
    public String getWorkerId() { return workerId; }
    public String getLabel() { return label; }
    public String getPortalUrl() { return portalUrl; }
}
